packages/di/src/common/interfaces/TokenProvider.ts

```
@@ -1,3 +1,4 @@
 import type {AbstractType, Type} from "@tsed/core";
 
-export type TokenProvider<T = any> = string | symbol | Type<T> | AbstractType<T> | Function;
+export type TokenWrapper<T = any> = T;
+export type TokenProvider<T = any> = string | symbol | Type<T> | AbstractType<T> | Function | TokenWrapper<T>;
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)